src/track/bpm.h

```diff
@@ -57,12 +57,34 @@ class Bpm final {
         m_value = kValueUndefined;
     }
 
+    void roundToInteger() {
+        if (!isValid()) {
+            return;
+        }
+        m_value = std::round(m_value);
+    }
+
+    QString displayString() {
+        return Bpm::displayString(m_value);
+    }
+
     static double valueFromString(const QString& str, bool* pValid = nullptr);
     static QString valueToString(double value);
     static int valueToInteger(double value) {
         return static_cast<int>(std::round(value));
     }
 
+    /// Returns a string depending on non zero decimal placess.
+    /// If the value is round enough, use 1 decimal places, otherwise 2
+    /// @param {value} bpm value
+    static QString displayString(double value) {
+        if (fabs(round(value * 10) / 10 - value) < 0.001) {
```

Comment:
Calculate the precision beforehand and then use a single, parameterized QString formatting expression. To reduce redundancy.